src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -245,7 +244,9 @@ protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions option
                 UserContext = options.UserContext,
 
                 Operation = operation,
-                Variables = variables,
+                Variables = validationResult.Variables ?? Variables.None,
+                ArgumentValues = validationResult.ArgumentValues,
```

Comment:
Design issue. Both `ArgumentValues` and `DirectiveValues` may be initialized here with some "null-object" like `Variables.None` or `Inputs.Empty`. All other collection/dictionary properties do so.

Reply:
I would prefer to leave as null when they have not been parsed.  If you agree, then they are nullable even when IsValid is true, and even if we set it to a null-like object such as Variables.None.  So then it seems there is little point to initializing to a null-like object.  Thoughts?

Reply:
> So then it seems there is little point to initializing to a null-like object.

Avoid null checks on client side? I'm fine to leave them as nullable if these properties are readonly by nature.  `OutputExtensions` and `Errors` (for example) are initialized with empty dictionary/collection because they get-only in `IExecutionContext` and caller should can change contents. If it is so then I would change type from `IDictionary` to `IReadonlyDictionary` to make explicit distinction beetween read-only properties and mutable properties of `IExecutionContext`.

Reply:
We could do `IReadOnlyDictionary<GraphQLField, IDictionary<string, ArgumentValue>>` of course...

Reply:
At least + place a general comment on `IExecutionContext` about distinction beetween read-only properties and mutable properties.